### PR TITLE
RDISCROWD-7338 Project password optional

### PR DESCRIPTION
**Pull request by n00rsy**
*Issue number of the reported bug or feature request: [RDISCROWD-7338](https://jira.prod.bloomberg.com/browse/RDISCROWD-7338)

**Describe your changes**
 - make project password optional instead of required

### CODE REVIEW

File: test/test_repository/test_project_repository.py [hunk 1]
```diff
@@ -183,12 +183,15 @@ def test_save(self):
 
     @with_context
     def test_save(self):
-        """Test save fails if the project has no password"""
+        """Test save does not fail if the project has no password"""
```
Comment (kbecker42) [inline test/test_repository/test_project_repository.py:186]: Should we include another test that checks if `PROJECT_PASSWORD_REQUIRED is True` then it should succeed?

File: test/test_repository/test_project_repository.py [hunk 2]
```diff
@@ -182,16 +183,29 @@ def test_save(self):
...
+    def test_save_project_password_not_required(self):
+        """Test save project without password success when password not required"""
+        with patch.dict(self.flask_app.config,
+                        {'PROJECT_PASSWORD_REQUIRED': False}):
+            project = ProjectFactory.build()
+            assert self.project_repo.get(project.id) is None
+
+            self.project_repo.save(project)
+
+            assert self.project_repo.get(project.id) == project, "Project not saved"
```
Comment (kbecker42) [inline test/test_repository/test_project_repository.py:208]: If the password is not required and not provided, shouldn't the project save succeed?

**Review by kbecker42 — APPROVED** [review]

File: pybossa/forms/forms.py [hunk 3]
```diff
@@ -99,7 +99,7 @@ class ProjectCommonForm(Form):
 
     password = PasswordField(
                     lazy_gettext('Password'),
-                    [validators.Required(),
+                    [validators.Optional(),
```
[the remark this answers is not in the document]
Comment (n00rsy) [reply pybossa/forms/forms.py]: updated via 8af49ce

**Review by kbecker42 — APPROVED** [review]

File: pybossa/view/projects.py [hunk 4]
```diff
@@ -949,6 +949,20 @@ def handle_valid_form(form):
                     prodsubprods=prodsubprods)
     return handle_content_type(response)
 
+
+@blueprint.route('/<short_name>/remove-password', methods=['POST'])
+@login_required
+@csrf.exempt
+def remove_password(short_name):
+    if current_app.config.get('PROJECT_PASSWORD_REQUIRED'):
```
Comment (kbecker42) [inline pybossa/view/projects.py:957]: The "remove password" link can also be hidden in the UI if this value is set in the config. See [here](https://github.com/bloomberg/pybossa-default-theme/pull/459/files#diff-39ee427448a2ae0a8931fd85fb0d77eccbbe4f22609a7af1d7533f7a8922a88eR66) and add a conditional similar to [this](https://github.com/bloomberg/pybossa-default-theme/blob/main/templates/projects/project.html#L67C21-L67C58).

```html
{% if config.PROJECT_PASSWORD_REQUIRED %}
  <form id="remove-password-form" method="post" action="{{ url_for('project.remove_password', short_name = project.short_name) }}">
      <div class="form-actions">
          <input type="hidden" name="csrf_token" value="{{ csrf_token() }}" />
          <input type="hidden" name="password" value="" />
          <button type="submit" class="btn btn-link" value="Remove Project Password">{{_('Remove Project Password')}}</button>
      </div>
  </form>
{% endif %}
```

Comment (n00rsy) [reply pybossa/view/projects.py:957]: implemented via b613336

**Review by kbecker42 — APPROVED** [review]

**Pull request merged**